src/utils/__tests__/async-test.js

```diff
@@ -1,14 +1,53 @@
 /* @flow strict-local */
 import { sleep, tryUntilSuccessful } from '../async';
+import { Lolex } from '../../__tests__/aux/lolex';
 
 describe('sleep', () => {
+  const lolex: Lolex = new Lolex();
+
+  afterAll(() => {
+    lolex.dispose();
+  });
+
+  afterEach(() => {
+    // clear any unset timers
+    lolex.clearAllTimers();
+  });
+
   test('waits for a given time in milliseconds', async () => {
     const expectedMs = 1000;
-    const start = Date.now();
-    await sleep(expectedMs);
-    const durationMs = Date.now() - start;
-    expect(expectedMs).toBeLessThanOrEqual(durationMs);
-    expect(durationMs).toBeLessThan(10 * expectedMs);
+
+    const sleepAsync = async () => {
+      const start = Date.now();
+      await sleep(expectedMs);
+      const actualMs = Date.now() - start;
+
+      return actualMs;
+    };
+    const sleepPromise: Promise<number> = sleepAsync();
+
+    // `sleepPromise` should have set up its timer before returning control.
+    expect(lolex.getTimerCount()).toBe(1);
+
+    const FUDGE_FACTOR = 1.01; // see discussion of [α], below
+    lolex.advanceTimersByTime(expectedMs * FUDGE_FACTOR);
```

Comment:
The FUDGE_FACTOR here seems like it makes this test loose in a way that it isn't now. I still don't see why any tradeoff like that should be a consequence of adopting a nice spiffy mock setup, where we get to control the clock's behavior in a way we didn't have before.

Reply:
Quite the opposite: it's intended to replicate the looseness that *is* present now. (Clarifying the preceding would also clarify this, I expect.)

I suppose we could have a separate test that's "sharp", so to speak. I'm not sure it would be terribly informative, though.